code/game/machinery/washing_machine.dm

```diff
@@ -14,306 +13,401 @@
+	door_open = !door_open
+	if(door_open)
+		eject_tub_contents()
+		update_washing_state()
+		return
+	update_washing_state()
+
+/// Attempts to locate stamps or crayons for dyeing purposes
+/obj/machinery/washing_machine/proc/locate_color_source()
+	var/list/color_sources = list()
+	for(var/obj/item/toy/crayon/C in contents)
+		color_sources += C
+	for(var/obj/item/stamp/S in contents)
+		color_sources += S
+	if(!length(color_sources))
+		return
+	color_source = pick(color_sources)
+	if(istype(color_source, /obj/item/toy/crayon))
+		var/obj/item/toy/crayon/C = color_source
+		wash_color = C.dye_color
+	if(istype(color_source, /obj/item/stamp))
+		var/obj/item/stamp/S = color_source
+		wash_color = S.dye_color
+
+/obj/machinery/washing_machine/proc/start_wash_cycle()
+	if(stat & (NOPOWER|BROKEN))
+		return
+	locate_color_source()
+	washing = TRUE
+	playsound(loc, 'sound/machines/click.ogg', 50, TRUE)
+	atom_say("Beginning Wash Cycle!")
+	sleep(1 SECONDS)
+	update_washing_state()
+	use_power(100)
+	playsound(loc, 'sound/effects/slosh.ogg', 25, TRUE)
+	addtimer(CALLBACK(src, PROC_REF(wash_cycle)), 15 SECONDS)
+
+/obj/machinery/washing_machine/proc/wash_cycle()
+	for(var/obj/item/I in inserted_items)
+		I.machine_wash(src)
+	for(var/mob/living/L in inserted_mobs)
+		L.machine_wash(src)
+	update_washing_state()
+	use_power(50 * current_tub_capacity)
+	playsound(loc, 'sound/effects/slosh.ogg', 25, TRUE)
+	addtimer(CALLBACK(src, PROC_REF(end_wash_cycle)), 15 SECONDS)
+
+/obj/machinery/washing_machine/proc/end_wash_cycle()
+	if(bloody_mess)
+		for(var/obj/item/I in inserted_items)
+			I.add_blood(I.blood_DNA)
+	if(color_source)
+		qdel(color_source)
+		color_source = null
+	playsound(loc, 'sound/effects/slosh.ogg', 25, TRUE)
+	washing = FALSE
+	atom_say("Wash Cycle Complete!")
+	sleep(1 SECONDS)
+	playsound(loc, 'sound/machines/ding.ogg', 50, TRUE)
+	update_washing_state()
+
+/obj/machinery/washing_machine/container_resist(mob/living/L)
+	. = ..()
+	if(washing)
+		return
+	if(!door_open)
+		toggle_door()
+		return
 	else
-		return ..()
+		L.forceMove(loc)
+		check_tub_contents()
 
-/obj/machinery/washing_machine/attack_hand(mob/user)
-	switch(state)
-		if(OPEN_EMPTY)
-			state = CLOSED_EMPTY
-		if(CLOSED_EMPTY)
-			for(var/atom/movable/O in src)
-				O.forceMove(loc)
-			crayon = null
-			state = OPEN_EMPTY
-		if(OPEN_FULL)
-			state = CLOSED_FULL
-		if(CLOSED_FULL)
-			for(var/atom/movable/O in src)
-				O.forceMove(loc)
-			crayon = null
-			state = OPEN_EMPTY
-		if(RUNNING)
-			to_chat(user, "<span class='warning'>[src] is busy.</span>")
-		if(CLOSED_BLOODY)
-			if(gibs_ready)
-				gibs_ready = FALSE
-				if(locate(/mob) in src)
-					var/mob/M = locate() in src
-					M.gib()
-			for(var/atom/movable/O in src)
-				O.forceMove(loc)
-			crayon = null
-			state = OPEN_EMPTY
-
-	update_icon(UPDATE_ICON_STATE)
+
+
+/obj/machinery/washing_machine/default_deconstruction_screwdriver(mob/user, icon_state_open, icon_state_closed, obj/item/I)
```

Comment:
IMO, This should never be overwritten. You should just set `icon_state_open` and `icon_state_closed` to the current icon_state (as much as I hate it.)

Reply:
removed in next commit

Reply:
updated in next commit